tensorflow_quantum/core/ops/tfq_utility_ops_test.py

```diff
@@ -126,5 +128,235 @@ def test_padded_to_ragged(self, padded_array):
+
+        # Get random circuit batches
+        qubits = cirq.GridQubit.rect(1, n_qubits)
+        batch_size = 15
+        n_moments = 15
+        circuit_batch, resolver_batch = \
+            util.random_symbol_circuit_resolver_batch(
+                qubits, symbol_names, batch_size, n_moments)
+
+        # Remove one of the symbols from the resolvers
+        symbol_names_partial = symbol_names[1:]
+        symbol_values_array_partial = np.array(
+            [[resolver[symbol]
+              for symbol in symbol_names_partial]
+             for resolver in resolver_batch])
+        resolver_batch_partial = [
+            cirq.ParamResolver(
+                {symbol: resolver[symbol]
+                 for symbol in symbol_names_partial})
+            for resolver in resolver_batch
+        ]
+
+        # Resolve in two ways and compare results
+        test_resolved_circuits = util.from_tensor(
+            tfq_utility_ops.resolve_parameters(
+                util.convert_to_tensor(circuit_batch), symbol_names_partial,
+                symbol_values_array_partial))
+        expected_resolved_circuits = []
+        for circuit, resolver in zip(circuit_batch, resolver_batch_partial):
+            expected_resolved_circuits.append(
+                cirq.resolve_parameters(circuit, resolver))
+        for test_c, exp_c in zip(test_resolved_circuits,
+                                 expected_resolved_circuits):
+            for test_m, exp_m in zip(test_c, exp_c):
+                for test_o, exp_o in zip(test_m, exp_m):
+                    tg = test_o.gate
+                    eg = exp_o.gate
+                    self.assertEqual(type(tg), type(eg))
+                    # TODO(zaqqwerty): simplify parsing when cirq build parser
+                    # see core/serialize/serializer.py
+                    if isinstance(tg, cirq.IdentityGate):
+                        # all identity gates are the same
+                        continue
+                    elif isinstance(tg, cirq.EigenGate):
+                        self._compare_gate_parameters(tg._global_shift,
+                                                      eg._global_shift)
+                        self._compare_gate_parameters(tg._exponent,
+                                                      eg._exponent)
+                    elif isinstance(tg, cirq.FSimGate):
+                        self._compare_gate_parameters(tg.theta, eg.theta)
+                        self._compare_gate_parameters(tg.phi, eg.phi)
+                    elif isinstance(
+                            tg, (cirq.PhasedXPowGate, cirq.PhasedISwapPowGate)):
+                        self._compare_gate_parameters(tg._global_shift,
+                                                      eg._global_shift)
+                        self._compare_gate_parameters(tg._exponent,
+                                                      eg._exponent)
+                        self._compare_gate_parameters(tg._phase_exponent,
```

Comment:
Can we not use gate equality checking with the `==` operation ?

Reply:
Gates are only approximately equal here (seems by difference of float and double); and, there was additionally difficulty getting expressions like `0.1*pi*a` and `0.314*a` to compare equal, which is taken care of in the sympy parsing at line 143

Reply:
maybe instead of having this intricate comparison function, we could just send both sides through the `convert_to_tensor` / `convert_from_tensor` pair until they equalize. In this case, something like:

```python3
expected_ans = tfq.from_tensor(tfq.to_tensor(resolved_circuits))
self.assertAllEqual(expected_ans, calculated_ans)
```

This way we add in all the same artifacts from going to and from tensor without having to have this equality checking code laying around.

Reply:
when I do this conversion equalizing, as
```python
tensor_test = util.from_tensor(util.convert_to_tensor(util.from_tensor(util.convert_to_tensor(test_resolved_circuits))))
tensor_expected = util.from_tensor(util.convert_to_tensor(util.from_tensor(util.convert_to_tensor(expected_resolved_circuits))))
self.assertAllEqual(tensor_test, tensor_expected)
```
For a batch and moment size of 2 in the circuit randomizer I get the error that the following terms are not equal:
```(cirq.XX**0.410076).on(cirq.GridQubit(0, 0), cirq.GridQubit(0, 1))```
```(cirq.XX**0.410075).on(cirq.GridQubit(0, 0), cirq.GridQubit(0, 1))```
Since the exponents are slightly different.  Using `self.assertAllClose` fails when encountering unresolved symbols.

Reply:
Trying rounding inside the util function now instead

Reply:
Did `cirq.approx_eq` also not work ?

Reply:
adding rounding to 3 decimal places to the symbol values in  `util.random_symbol_circuit_resolver_batch` did not help resolve this either, added back the parsing for now